File: Design_Pattern_and_Principles/CommandPatternExample/CommandTest.java
public class CommandTest {
    public static void main(String[] args) {
        Light livingRoomLight = new Light();
        RemoteControl remote = new RemoteControl();

        Command on = new LightOnCommand(livingRoomLight);
        Command off = new LightOffCommand(livingRoomLight);
        remote.assignCommand(on);
        remote.pressButton();
        remote.assignCommand(off);
        remote.pressButton();
    }
}

File: Week_1/Design_Pattern_and_Principles/CommandPatternExample/CommandTest.cs
public class CommandTest {
    public static void Main(string[] args) {
        Light livingRoomLight = new Light();
        RemoteControl remote = new RemoteControl();

        ICommand on = new LightOnCommand(livingRoomLight);
        ICommand off = new LightOffCommand(livingRoomLight);

        remote.AssignCommand(on);
        remote.PressButton();

        remote.AssignCommand(off);
        remote.PressButton();
    }
}